Compiler/script2/cs_scanner.cpp

```diff
@@ -544,14 +545,34 @@ AGS::ErrorType AGS::Scanner::ReadInStringLit(std::string &symstring, std::string
             ErrorType retval = SkipWhitespace();
             if (retval < 0) return retval;
 
-            if ('"' != Peek())
+
+            if ('"' != Peek() ||
+                kNewSectionLitPrefix == valstring.substr(0, kNewSectionLitPrefix.length()))
             {
                 _inputStream.seekg(pos_before_skip);
                 _lineno = lineno_before_skip;
-                return kERR_None; // End of string
+                return kERR_None;
             }
 
             // Another string literal follows
+            // Tentatively read ahead to check whether it's a new section marker
+            pos_before_skip = _inputStream.tellg();
+            lineno_before_skip = _lineno;
+            Get(); // Eat leading '"'
+            char *tbuffer = new char[kNewSectionLitPrefix.length() + 1];
```

Comment:
Randomly noticed this, I think allocating memory could be avoided since kNewSectionLitPrefix is a constant string.
Even if you cannot use its "length()" there because its not `constexpr` (and you don't want to make this string a literal), you still could use a buffer of large enough fixed size to accomodate it.